### PR TITLE
feat: create tokens page structure

### PR DESCRIPTION
**Pull request by lucas3003**
### Acceptance Criteria

The following UI components must be created. Real data is not required at this moment:

- Navigation link
- Table
- Components
- Searching/pagination

### Testing

These are the performed tests (Using mocked data):

#### Desktop:

https://user-images.githubusercontent.com/3287486/160186605-96a07b05-0606-4643-8948-bf75832a5593.mov

When no Token is found:

Screen Shot 2022-03-25 at 16 08 57

#### Mobile horizontal:

https://user-images.githubusercontent.com/3287486/160186719-60dcdc6b-de2d-4649-a0c1-358f684f8118.mov

#### Mobile vertical:

https://user-images.githubusercontent.com/3287486/160186674-7e5668f5-a2d6-4238-85b6-bb0aeff8e992.mov

### CODE REVIEW

**Review by r4mmer — CHANGES_REQUESTED** [review]
I think we should move the `Tokens` component content to the `TokenList` screen and make a `TokenTable` component which is more reusable than the `TokenRow`.

What do you think?

File: src/components/token/Tokens.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import colors from '../../index.scss';
+import TokenRow from './TokenRow';
+import {
+    shouldRenderCustomTokens
+} from '../../feature';
+
+/**
+ * Displays custom tokens in a table with pagination buttons and a search bar.
+ */
+class Tokens extends React.Component {
```
Comment (pedroferreira1) [inline src/components/token/Tokens.js:20]: I have the same comment about the prop types here

File: src/components/token/Tokens.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            if (this.state.tokens.length === 0) {
+                return null;
+            } else {
+                return (
+                    <nav aria-label="Token pagination" className="d-flex justify-content-center">
+                        <ul className="pagination">
+                            <li ref="tokenPrevious" className={(!this.state.hasBefore || this.state.tokens.length === 0) ? "page-item mr-3 disabled" : "page-item mr-3"}>
+                                <Link to={this} className="page-link">Previous</Link>
+                            </li>
+                            <li ref="tokenNext" className={(!this.state.hasAfter || this.state.tokens.length === 0) ? "page-item disabled" : "page-item"}>
```
Comment (pedroferreira1) [inline src/components/token/Tokens.js:45]: You are handling `this.state.tokens.length === 0` in the first condition of the `if`, right?

File: src/components/token/Tokens.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                    <TokenRow key={token.id} token={token} />
+                );
+            });
+        }
+
+
+        const loadTable = () => {
+            return (
+                <div className="table-responsive mt-2">
+                    <table className="table table-striped" id="tx-table">
```
[the remark this answers is not in the document]
Comment (lucas3003) [reply src/components/token/Tokens.js]: I created a new component called TokensTable that I believe will modularize better.

File: src/components/token/TokenRow.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { withRouter } from "react-router-dom";
+
+class TokenRow extends React.Component {
```
[the remark this answers is not in the document]
Comment (lucas3003) [reply src/components/token/TokenRow.js:13]: Added PropTypes in the new commit.

File: src/components/token/Tokens.js (hunk 1 above)
Comment (lucas3003) [reply src/components/token/Tokens.js:19]: Added PropTypes in the new commit.

File: src/components/token/Tokens.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        const loadTableBody = () => {
+            return this.state.tokens.map((token, idx) => {
+                return (
+                    <TokenRow key={token.id} token={token} />
+                );
+            });
+        }
+
+
+        const loadTable = () => {
```
[the remark this answers is not in the document]
Comment (lucas3003) [reply src/components/token/Tokens.js]: I added the message on the new commit and changed the description showing how it looks like.

File: src/components/token/Tokens.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                    </table>
+                </div>
+            );
+        }
+
+        const loadSearchField = () => {
+            return (
+                <div className="d-flex flex-row align-items-center navigation-search-token">
+                    <div className="d-flex flex-row align-items-center col-12">
+                        <input className="form-control mr-2" type="search" placeholder="Search ID, name, symbol, or type" aria-label="Search" ref="tokenSearch" />
```
[the remark this answers is not in the document]
Comment (lucas3003) [reply src/components/token/Tokens.js]: Changed in the new commit.

File: src/components/token/Tokens.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                            {loadTableBody()}
+                        </tbody>
+                    </table>
+                </div>
+            );
+        }
+
+        const loadSearchField = () => {
+            return (
+                <div className="d-flex flex-row align-items-center navigation-search-token">
```
[the remark this answers is not in the document]
Comment (lucas3003) [reply src/components/token/Tokens.js]: Addressed these 3 point in the new commit. Also, changed the description with the new UI

File: src/components/token/TokensTable.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+                        </ul>
+                    </nav>
+                );
+            }
+        }
+
+        const loadTableBody = () => {
+            return this.props.tokens.map((token, idx) => {
+                return (
+                    <TokenRow key={token.id} token={token} />
```
Comment (pedroferreira1) [inline src/components/token/TokensTable.js:30]: The proptypes of `TokenRow` uses `token.uid` instead of `token.id`

**Review by pedroferreira1 — APPROVED** [review]

**Review by r4mmer — APPROVED** [review]

**Pull request merged**